mirgecom/profiling.py

```diff
@@ -31,11 +31,96 @@
 import numpy as np
 from dataclasses import dataclass
 import pytools
+from logpyle import LogManager
+from mirgecom.logging_quantities import KernelProfile
+from statistics import mean
 
 __doc__ = """
 .. autoclass:: PyOpenCLProfilingArrayContext
+.. autofunction:: add_nonloopy_profiling_result
 """
 
+# {{{ Support for non-Loopy results (e.g., pyopencl kernels)
+
+nonloopy_profile_results = {}
+
+
+def add_nonloopy_pyopencl_results(queue, events: list) -> None:
+    """Gather and store pyopencl.array results."""
+    if queue and \
+      queue.properties & cl.command_queue_properties.PROFILING_ENABLE:
+
+        times = []
+        for evt in events:
+            time = evt.profile.end - evt.profile.start
+            times.append(time)
+
+        add_nonloopy_profiling_result("pyopencl_array", mean(times))
+
+
+def pyopencl_monkey_del(self):
+    """Monkey patches ``pyopencl.array.Array.__del__``  to grab profile data."""
+    if self.events:
+        cl.wait_for_events(self.events)
+
+        add_nonloopy_pyopencl_results(self.queue, self.events)
+
+        del self.events[:]
```

Comment:
I think we'll actually get better/higher-fidelity information if we monkey-patch `pyopencl.array.elwise_kernel_runner`. At least we'll be able to get bandwidth info by assuming that each array involved gets read/written once.